pkg/serverless/protocol.go

```diff
@@ -397,8 +399,9 @@ type Flush struct {
 
 // ServeHTTP - see type Flush comment.
 func (f *Flush) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	f.daemon.flushing.Lock()
```

Comment:
This method only applies to the /flush API call. I think it makes more sense for the mutex lock to happen inside the 'Daemon.TriggerFlush' method, since other parts of the code can try and call that method, and it's what we are trying to protect.